components/properties/src/script.rs

```diff
@@ -140,6 +165,65 @@ impl<'data> ScriptExtensions<'data> {
             ZeroSlice::from_ule_slice(script_ule_slice)
         }
     }
+
+    /// Return the `Script_Extensions` property value for this code point.
+    ///
+    /// If `code_point` has Script_Extensions, then return the Script codes in
+    /// the Script_Extensions. In this case, the Script property value
+    /// (normally Common or Inherited) is not included in the set.
+    ///
+    /// If c does not have Script_Extensions, then the one Script code is put
+    /// into the set and also returned.
+    ///
+    /// If c is not a valid code point, then the one UNKNOWN code is put into
+    /// the set and also returned.
+    pub fn get_script_extensions_val(&self, code_point: u32) -> &ZeroSlice<Script> {
+        let sc_with_ext = self.trie.get(code_point);
+
+        self.get_scx_val_using_trie_val(code_point, sc_with_ext)
+    }
+
+    /// Returns whether `script` is contained in the Script_Extensions
+    /// property value if the code_point has Script_Extensions, otherwise
+    /// if the code point does not have Script_Extensions then returns
+    /// whether the Script property value matches.
+    ///
+    /// Some characters are commonly used in multiple scripts. For more information,
+    /// see UAX #24: http://www.unicode.org/reports/tr24/.
+    pub fn has_script(&self, code_point: u32, script: Script) -> bool {
+        let sc_with_ext = self.trie.get(code_point);
+
+        if !sc_with_ext.has_extensions() {
+            let script_val = sc_with_ext.0;
+            script == Script(script_val)
+        } else {
+            let scx_val = self.get_scx_val_using_trie_val(code_point, sc_with_ext);
+            let script_find = scx_val.iter().find(|&sc| sc == script);
+            script_find.is_some()
+        }
+    }
+
+    /// Returns a [`UnicodeSet`] for the given [`Script`] which represents all
+    /// code points for which `has_script` will return true.
+    pub fn get_script_extensions_set(&self, script: Script) -> UnicodeSet {
+        let ranges_matching_script = self.trie.iter_ranges().filter(|cpm_range| {
+            let start = cpm_range.get_start();
+            let sc_with_ext = ScriptWithExt(cpm_range.get_value() as u16);
+
+            (!sc_with_ext.has_extensions() && script == sc_with_ext.into())
+                || (sc_with_ext.has_extensions()
```

Comment:
Can you use a ternary operator here so that you need to test sc_with_ext.has_extensions() only once?